### PR TITLE
Sudoku Solved :)

### PR DESCRIPTION
**Pull request by DeepthiMuppala**
Implemented Sudoku game

Reviewers:
@ravichandrasadineni

### CODE REVIEW

File: games/sudoku-solver/Main.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Main {
+  public void playSudoku(Sudoku sudoku) {
+    int[][] board = new int[9][9];
+    int randomRow = (int) Math.ceil(Math.random() * 9) - 1;
+    int randomCol = (int) Math.ceil(Math.random() * 9) - 1;
+    int randomVal = (int) Math.ceil(Math.random() * 9) - 1;
```
Comment (ravichandrasadineni) [inline games/sudoku-solver/Main.java]: Should this be 1-9 instead of 0-8

File: games/sudoku-solver/Main.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Main {
+  public void playSudoku(Sudoku sudoku) {
+    int[][] board = new int[9][9];
```
Comment (ravichandrasadineni) [inline games/sudoku-solver/Main.java]: Shouldn't this and next 4 lines be part of Sudoku constructor. Why are we doing these here ?

File: games/sudoku-solver/Main.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    int randomCol = (int) Math.ceil(Math.random() * 9) - 1;
+    int randomVal = (int) Math.ceil(Math.random() * 9) - 1;
+    board[randomRow][randomCol] = randomVal;
+    sudoku.sudokuSolver(board);
+  }
+
+  public static void main(String[] args) {
+    Sudoku sudoku = new Sudoku();
+    Main m = new Main();
+    m.playSudoku(sudoku);
```
Comment (ravichandrasadineni) [inline games/sudoku-solver/Main.java]: nit: Rename this to generateNewBoard. We are not actually playing here.

File: games/sudoku-solver/Sudoku.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import java.util.*;
+class Sudoku {
+  public void sudokuSolver(int[][] board) {
```
Comment (ravichandrasadineni) [inline games/sudoku-solver/Sudoku.java]: Do not pass board from the Main. Let it be part of the this class.  Random initialization of the board should be part of this class.

Comment (ravichandrasadineni) [inline games/sudoku-solver/Sudoku.java]: nit : poor naming. Please rename this to generateBoard()

File: games/sudoku-solver/Sudoku.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    if(board.length == 0) return;
+    sudokuSolverHelper(board);
+    printBoard(board);
```
Comment (ravichandrasadineni) [inline games/sudoku-solver/Sudoku.java]: Do not print board as part of this function. Expose this function outside and call it directly from main.

File: games/sudoku-solver/Sudoku.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    return true;
+  }
+
+  public boolean isValid(int i, int j, int[][] board, int val) {
+    for (int t = 0; t < board[0].length; t++) {
+      if (board[i][t] != 0 && board[i][t] == val) {
+        return false;
+      } else if (board[t][j] != 0 && board[t][j] == val) {
+        return false;
+      } else if (board[3 * (i / 3) + t / 3][3 * (j / 3) + t % 3] != 0 && board[3 * (i / 3) + t / 3][3 * (j / 3) + t % 3] == val)
```
Comment (ravichandrasadineni) [inline games/sudoku-solver/Sudoku.java]: Poorly indented.

File: games/sudoku-solver/Main.java (hunk 1 above)
Comment (DeepthiMuppala) [reply games/sudoku-solver/Main.java]: Agree, fixed here e18c92a

File: games/sudoku-solver/Main.java (hunk 2 above)
Comment (DeepthiMuppala) [reply games/sudoku-solver/Main.java]: Agree, fixed here e18c92a

File: games/sudoku-solver/Main.java (hunk 3 above)
Comment (DeepthiMuppala) [reply games/sudoku-solver/Main.java]: e18c92a

File: games/sudoku-solver/Sudoku.java (hunk 4 above)
Comment (DeepthiMuppala) [reply games/sudoku-solver/Sudoku.java]: e18c92a

File: games/sudoku-solver/Sudoku.java (hunk 5 above)
Comment (DeepthiMuppala) [reply games/sudoku-solver/Sudoku.java]: Agree, fixed here e18c92a

File: games/sudoku-solver/Sudoku.java (hunk 4 above)
Comment (DeepthiMuppala) [reply games/sudoku-solver/Sudoku.java]: Agree, fixed here e18c92a

File: games/sudoku-solver/Sudoku.java (hunk 6 above)
Comment (DeepthiMuppala) [reply games/sudoku-solver/Sudoku.java]: done e18c92a

**Review by ravichandrasadineni — APPROVED** [review]

**Pull request merged**